numpy/core/src/multiarray/convert_datatype.c

```diff
@@ -1700,13 +1702,13 @@ PyArray_ResultType(
                 if (tmp == NULL) {
                     goto error;
                 }
-                curr = common_dtype->discover_descr_from_pyobject(common_dtype, tmp);
+                curr = NPY_DT_call_discover_descr_from_pyobject(common_dtype, tmp);
                 Py_DECREF(tmp);
             }
             if (curr == NULL) {
                 goto error;
             }
-            Py_SETREF(result, common_dtype->common_instance(result, curr));
+            Py_SETREF(result, NPY_DT_SLOTS(common_dtype)->common_instance(result, curr));
```

Comment:
Should this also be a macro? `NPY_DT_common_instance(common_dtype, result, curr)`?

Reply:
I am ambivalent, I did not do it, since it the actual arguments here are just the instances (and `NPY_DT_common_instance(common_dtype)(result, curr)` seemed even weirder.

Reply:
OK, no problem.